src/core/tests/replace_node.cpp

```diff
@@ -128,6 +128,31 @@ TEST(replace_node, simple_node_replacement) {
     ASSERT_EQ(new_relu->output(0).get_tensor().get_names(), std::unordered_set<std::string>({"c", "d", "f"}));
 }
 
+TEST(replace_node, replacement_with_direct_parent_node) {
+    auto param = std::make_shared<ov::op::v0::Parameter>(element::i64, Shape{1, 64});
+    param->output(0).get_tensor().set_names({"a", "b"});
+
+    auto child_1 = std::make_shared<ov::op::v0::Relu>(param);
+    auto child_2 = std::make_shared<ov::op::v0::Relu>(param);
+
+    auto model = std::make_shared<ov::Model>(OutputVector{child_1, child_2}, ParameterVector{param});
+    EXPECT_NO_THROW(model->validate_nodes_and_infer_types());
+
+    auto relu = std::make_shared<ov::op::v0::Relu>(param);
+    relu->output(0).get_tensor().set_names({"c", "d"});
+    replace_node(param, relu);
+
+    EXPECT_NO_THROW(model->validate_nodes_and_infer_types());
```

Comment:
```suggestion
    ASSERT_NO_THROW(model->validate_nodes_and_infer_types());
```
Consider add comment that this check validates that loop between input - output is not created?

Reply:
done